common.js

```diff
@@ -1728,6 +1728,586 @@ const allSoloCars = {
+      '1989': ['dp'],
+      '1990': ['dp'],
+      '1991': ['dp'],
+      '1992': ['dp'],
+      '1993': ['dp'],
+      '1994': ['dp'],
+      '1995': ['dp'],
+      '1996': ['dp'],
+      '1997': ['dp'],
+      '1998': ['dp']
+    },
+    '240SX': {
+      'all': ['hs', 'stx']
+    },
+    '240Z': {
+      'all': ['bsp']
+    },
+    '240Z (incl. 2+2) - Alternate part: headlight covers': {
+      '1970': ['fp'],
+      '1971': ['fp'],
+      '1972': ['fp'],
+      '1973': ['fp'],
+      '1974': ['fp'],
+      '1975': ['fp'],
+      '1976': ['fp'],
+      '1977': ['fp'],
+      '1978': ['fp']
+    },
+    '260Z': {
+      'all': ['bsp']
+    },
+    '260Z (incl. 2+2) - Alternate part: headlight covers': {
+      '1970': ['fp'],
+      '1971': ['fp'],
+      '1972': ['fp'],
+      '1973': ['fp'],
+      '1974': ['fp'],
+      '1975': ['fp'],
+      '1976': ['fp'],
+      '1977': ['fp'],
+      '1978': ['fp']
+    },
+    '280Z': {
+      'all': ['bsp']
+    },
+    '280Z (incl. 2+2) - Alternate part: headlight covers': {
+      '1970': ['fp'],
+      '1971': ['fp'],
+      '1972': ['fp'],
+      '1973': ['fp'],
+      '1974': ['fp'],
+      '1975': ['fp'],
+      '1976': ['fp'],
+      '1977': ['fp'],
+      '1978': ['fp']
+    },
+    '280ZX': {
+      'all': ['bsp']
+    },
+    '280ZX (incl. 2+2) - Alternate part: headlight covers': {
+      '1979': ['fp'],
+      '1980': ['fp'],
+      '1981': ['fp'],
+      '1982': ['fp'],
+      '1983': ['fp']
+    },
+    '280ZX Turbo': {
+      'all': ['bsp']
+    },
+    '300ZX (non-turbo)': {
+      '1984': ['hs', 'stu', 'stx', 'esp'],
+      '1985': ['hs', 'stu', 'stx', 'esp'],
+      '1986': ['hs', 'stu', 'stx', 'esp'],
+      '1987': ['hs', 'stu', 'stx', 'esp'],
+      '1988': ['hs', 'stu', 'stx', 'esp'],
+      '1989': ['hs', 'stu', 'stx', 'esp'],
+      '1990': ['fs', 'esp'],
+      '1991': ['fs', 'esp'],
+      '1992': ['fs', 'esp'],
+      '1993': ['fs', 'esp'],
+      '1994': ['fs', 'esp'],
+      '1995': ['fs', 'esp'],
+      '1996': ['fs', 'esp']
+    },
+    '300ZX Turbo': {
+      '1984': ['fs', 'bsp'],
+      '1985': ['fs', 'bsp'],
+      '1986': ['fs', 'bsp'],
+      '1987': ['fs', 'bsp'],
+      '1988': ['fs', 'bsp'],
+      '1989': ['fs', 'bsp'],
+      '1990': ['cs', 'bsp'],
+      '1991': ['cs', 'bsp'],
+      '1992': ['cs', 'bsp'],
+      '1993': ['cs', 'bsp'],
+      '1994': ['cs', 'bsp'],
+      '1995': ['cs', 'bsp'],
+      '1996': ['cs', 'bsp']
+    },
+    '300ZX (Z31 chassis) - Alternate part: headlight covers': {
+      '1984': ['fp'],
+      '1985': ['fp'],
+      '1986': ['fp'],
+      '1987': ['fp'],
+      '1988': ['fp'],
+      '1989': ['fp']
+    },
+    '300ZX (Z32 chassis; non-turbo) - Alternate part: rear facing hood scoop (3.5” max height)': {
+      '1990': ['fp'],
+      '1991': ['fp'],
+      '1992': ['fp'],
+      '1993': ['fp'],
+      '1994': ['fp'],
+      '1995': ['fp'],
+      '1996': ['fp']
+    },
+    '310': {
+      'all': ['fsp']
+    },
+    '350Z': {
```

Comment:
This should be merged with the `350Z (non-NISMO)` that you've got below, simply by adding `fp` and `bsp` to all model years. Also, please remove the `all` if you do specify a year, it is only meant to cover the cases where a year/range is not specified so having both `all` and specific years is a bit confusing. Also, the 350Z was produced from 2002 to 2009 according to [wikipedia](https://en.wikipedia.org/wiki/Nissan_350Z) so that's the year range I recommend.

Reply:
Addressed in the following commit:
https://github.com/Bjorn248/scca_classifier/pull/81/commits/8f23201267adb4632a31083a697460065a84ef6a